lib/format_dpdk.c

```diff
@@ -1376,6 +1418,11 @@ static int dpdk_start_streams(struct dpdk_format_data_t *format_data,
 		stream = libtrace_list_get_index(format_data->per_stream, i)->data;
 		stream->queue_id = i;
 
+                /* Update the core with the user supplied core. User supplied cores are
+                 * -1 by default so if a core is not specific this will have no effect.
+                 */
+                stream->lcore = format_data->cores[i];
```

Comment:
I think this also should set (and check it was not already allocated) core_config[i] = 1;
See the logic in dpdk_reserve_lcore()
Otherwise, unspecified cores might pick the same core as a user pinned one.

Reply:
Have now done this in dpdk_reserve_lcore() so the core is also reserved [format_dpdk.c](https://github.com/LibtraceTeam/libtrace/blob/6300f271278fd3ca2e1f80f11163a5629270baac/lib/format_dpdk.c#L1014-L1019)